src/ui/public/doc_table/doc_table.less

```diff
@@ -43,11 +47,13 @@ doc-table {
         background-color: @panel-bg;
         white-space: nowrap;
         right: 0;
+        opacity: 0; /* 1 */
       }
 
+      &:focus,
       &:hover {
-        .docTableRowFilterButton {
-          opacity: 1;
+        .table-cell-filter {
+          opacity: 1; /* 2 */
```

Comment:
Unfortunately, if you tab to the buttons, they won't become visible because these styles only apply when the focus state is on the parent. I think the best solution is to remove the `background-color: @panel-bg;` property from line 43, and change `.docTableRowFilterButton` to have these styles:

```less
/**
 * 1. Align icon with text in cell.
 * 2. Use opacity to make this element accessible to screen readers and keyboard.
 * 3. Show on focus to enable keyboard accessibility.
 */
.docTableRowFilterButton {
  appearance: none;
  background-color: @panel-bg;
  border: none;
  padding: 0 3px;
  font-size: 14px;
  line-height: 1; /* 1 */
  display: inline-block;
  opacity: 0; /* 2 */

  &:focus {
    opacity: 1; /* 3 */
  }
}
```

This way the buttons themselves have the background color, so it will only obscure the text when the buttons themselves are focused or the row is being hovered over.